src/NuGet.Services.BasicSearch/SecretReader/ISecretReaderFactory.cs

```diff
@@ -0,0 +1,15 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using NuGet.Indexing;
+using NuGet.Services.KeyVault;
+
+namespace NuGet.Services.BasicSearch
+{
+    internal interface ISecretReaderFactory
+    {
+        ISecretInjector CreateSecretInjector(ISecretReader secretReader);
+
+        ISecretReader CreateSecterReader(IConfiguration configuration);
```

Comment:
good catch. Thanks :)

---
In reply to: [71401074](https://github.com/NuGet/NuGet.Services.Metadata/pull/78#discussion_r71401074) [](ancestors = 71401074)